frappe/public/js/frappe/form/controls/base_input.js

```diff
@@ -83,7 +83,9 @@ frappe.ui.form.ControlInput = class ControlInput extends frappe.ui.form.Control
 				me.value = me.doc[me.df.fieldname] || "";
 			}
 
-			if (me.can_write()) {
+			let is_fetch_from_read_only = me.df.fetch_from && !me.df.fetch_if_empty;
```

Comment:
@shariquerik Shouldn't this be read **only** if the field it's fetching from is set. If fetch_from is set but the link field is empty, the user can still modify the field

Reply:
Nevermind, fixed in https://github.com/frappe/frappe/pull/19192